isa/rv64si/csr.S

```diff
@@ -81,6 +81,22 @@ RVTEST_CODE_BEGIN
   andi a0, a0, 1  # a0 = a0 & 1
   beqz a0, finish # if no user mode, skip the rest of these checks
 #endif /* __MACHINE_MODE */
+  la t0, user_mode_end
+  srli t0, t0, 2
```

Comment:
To improve the readability of this code, please use the PMP_SHIFT macro, instead of the magic constant 2.

You can see the PMP macros here: https://github.com/riscv/riscv-test-env/blob/master/encoding.h#L135